lib/improver/between_thresholds.py

```diff
@@ -0,0 +1,205 @@
+
+    @staticmethod
+    def _get_multiplier(thresh_coord):
+        """
+        Check whether the cube contains "above" or "below" threshold
+        probabilities.  For "above", the probability of occurrence between
+        thresholds is the difference between probabilities at the lower
+        and higher thresholds: P(lower) - P(higher).  For "below" it is the
+        inverse of this: P(higher) - P(lower), which is implemented by
+        multiplying the difference by -1.
+
+        Args:
+            thresh_coord (iris.coords.DimCoord):
+                Threshold-type coordinate from the input cube
+
+        Returns:
+            multiplier (float):
+                1. or -1.
+
+        Raises:
+            ValueError: If the spp__relative_to_threshold attribute is
+                not recognised
+        """
+        if thresh_coord.attributes['spp__relative_to_threshold'] == 'above':
+            multiplier = 1.
+        elif thresh_coord.attributes['spp__relative_to_threshold'] == 'below':
+            multiplier = -1.
+        else:
+            raise ValueError('Input cube must contain probabilities of '
+                             'occurrence above or below threshold')
+        return multiplier
+
+    def _calculate_probabilities(self, thresh_coord):
+        """
+        Calculate between_threshold probabilities cube
+
+        Args:
+            thresh_coord (iris.coords.DimCoord):
+                Threshold-type coordinate from the input cube
+
+        Returns:
+            (iris.cube.Cube):
+                Merged cube containing recalculated probabilities
+        """
+        multiplier = self._get_multiplier(thresh_coord)
+        thresh_name = thresh_coord.name()
+
+        cubelist = iris.cube.CubeList([])
+        for (lower_cube, upper_cube) in self.cube_slices:
+            # construct difference cube
+            between_thresholds_data = (
+                lower_cube.data-upper_cube.data)*multiplier
+            between_thresholds_cube = upper_cube.copy(between_thresholds_data)
+
+            # add threshold coordinate bounds
+            lower_threshold = lower_cube.coord(thresh_name).points[0]
+            upper_threshold = upper_cube.coord(thresh_name).points[0]
+            between_thresholds_cube.coord(thresh_name).bounds = (
+                [lower_threshold, upper_threshold])
+
+            cubelist.append(between_thresholds_cube)
+
+        return cubelist.merge_cube()
+
+    def process(self, cube):
+        """
+        Calculate probabilities between thresholds for the input cube
+
+        Args:
+            cube (iris.cube.Cube):
+                Probability cube containing thresholded data (above or below)
+
+        Returns:
+            output_cube (iris.cube.Cube):
+                Cube containing probability of occurrence between thresholds
+        """
+        # if cube has no threshold-type coordinate, raise an error
+        try:
+            thresh_coord = find_threshold_coordinate(cube)
```

Comment:
I need to interrogate the coordinate attributes for `_get_multiplier`, which is why I have set this variable as the whole coordinate rather than the name.  However it does make sense to make this a class attribute.  I'll update this.